### PR TITLE
ARROW-13904: [R] Implement ModeOptions

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

File: r/src/compute.cpp [hunk 1]
```diff
@@ -398,6 +398,21 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
     return out;
   }
 
+  if (func_name == "mode") {
+    using Options = arrow::compute::ModeOptions;
+    auto out = std::make_shared<Options>(Options::Defaults());
+    if (!Rf_isNull(options["n"])) {
+      out->n = cpp11::as_cpp<int>(options["n"]);
```
Comment (pitrou) [inline r/src/compute.cpp:405]: `ModeOptions::n` is an `int64_t`, so perhaps use `cpp11::as_cpp<int64_t>`?

File: r/src/compute.cpp [hunk 2]
```diff
@@ -398,6 +398,21 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
...
+    }
+    if (!Rf_isNull(options["min_count"])) {
+      out->min_count = cpp11::as_cpp<int>(options["min_count"]);
```
Comment (pitrou) [inline r/src/compute.cpp:408]: For the same reason, perhaps `cpp11::as_cpp<uint32_t>`.

**Review by pitrou — COMMENTED** [review]
A quick review on the C++ part. I won't comment on the R tests.

File: r/src/compute.cpp (hunk 1 above)
Comment (thisisnic) [reply r/src/compute.cpp:405]: Cheers for spotting that, will update them both!

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**